streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3491,8 +3491,8 @@ final class ZStream[-R, +E, +A] private (val channel: ZChannel[R, Any, Any, Any,
   def toPull(implicit trace: Trace): ZIO[R with Scope, Nothing, ZIO[R, Option[E], Chunk[A]]] =
     channel.toPull.map { pull =>
       pull.mapError(error => Some(error)).flatMap {
-        case Left(done)  => ZIO.fail(None)
-        case Right(elem) => ZIO.succeed(elem)
+        case Left(done)  => Exit.failNone
```

Comment:
Let's use `foldZIO` instead of `mapError` and `flatMap` in the line above this while we're at it

Reply:
Done